internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixInstanceConfigRepository.java

```diff
@@ -89,6 +91,10 @@ public int getInstanceGroupId(String instanceId) {
     return groupId;
   }
 
+  public Map<String, Integer> getInstanceGroupIdMapping() {
+    return unmodifiableInstanceGroupIdMapping;
```

Comment:
Tried to follow encapsulation here by using a read only wrapper for the map. It's re-assigned to an updated map inside `instanceGroupIdMapping` in advance rather than initializing one here

Reply:
Makes sense, removed the mutable map altogether